flink-ml-lib/src/main/java/org/apache/flink/ml/common/param/linear/HasBatchSize.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.ml.common.param.linear;
+
+import org.apache.flink.ml.param.IntParam;
+import org.apache.flink.ml.param.Param;
+import org.apache.flink.ml.param.ParamValidators;
+import org.apache.flink.ml.param.WithParams;
+
+/** Interface for the shared batchSize param. */
+public interface HasBatchSize<T> extends WithParams<T> {
```

Comment:
Could you explain why we need this batch size parameter but Spark's logistic regression does not?

Reply:
Batch size is a terminology/concept in machine learning. Please also refer to Sklearn/Tensorflow/PyTorch.

Spark used `miniBatchFraction` because it is expensive to sample a fixed number of data points if you can only access the data once. It is a compromise for performance.

But in FlinkML, we have the opportunity to cache data by ourselves efficiently. Thus we should probably go back to `batch size.`